sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/CustomizingStatementHandler.java

```diff
@@ -56,15 +57,21 @@
         this.sqlObjectType = type;
         this.method = method;
 
-        // Prepare customizers that don't depend on actual arguments.
-        final Stream<BoundCustomizer> methodCustomizers = concat(
-                annotationsFor(type).map(a -> instantiateFactory(a).createForType(a, type)),
-                annotationsFor(method).map(a -> instantiateFactory(a).createForMethod(a, type, method)))
+        // Include annotations on the interface's supertypes
+        final Stream<BoundCustomizer> typeCustomizers = concat(Stream.of(type.getInterfaces()), Stream.of(type))
+            .flatMap(t -> annotationsFor(t))
+            .map(a -> instantiateFactory(a).createForType(a, type))
             .map(BoundCustomizer::of);
 
-        // Append customizers that do.
-        statementCustomizers = concat(methodCustomizers, parameterCustomizers(type, method))
-            .collect(Collectors.<BoundCustomizer>toList());
+        final Stream<BoundCustomizer> methodCustomizers =  annotationsFor(method)
+            .map(a -> instantiateFactory(a).createForMethod(a, type, method))
+            .map(BoundCustomizer::of);
+
+        final Stream<BoundCustomizer> parameterCustomizers = parameterCustomizers(type, method);
+
+        statementCustomizers = Stream.of(typeCustomizers, methodCustomizers, parameterCustomizers)
+            .flatMap(Function.identity())
```

Comment:
For my personal taste, `.reduce(Stream.empty(), Stream::concat)` looks more natural.